docling_core/types/doc/document.py

```diff
@@ -1297,17 +1581,241 @@ def iterate_items(
                         page_no=page_no,
                     )
 
+    def clear_cache(self):
+        """Clear cache storage of all images."""
+        for ix, (item, level) in enumerate(self.iterate_items(with_groups=True)):
+            if isinstance(item, PictureItem):
+                if item.image is not None and item.image._pil is not None:
+                    item.image._pil.close()
+
+    def _list_images_on_disk(self) -> List[Path]:
+        """List all images on disk."""
+        result: List[Path] = []
+
+        for ix, (item, level) in enumerate(self.iterate_items(with_groups=True)):
+            if isinstance(item, PictureItem):
+                if item.image is not None and isinstance(item.image.uri, Path):
+                    result.append(item.image.uri)
+
+                elif (
+                    item.image is not None
+                    and isinstance(item.image.uri, AnyUrl)
+                    and str(item.image.uri).startswith("file:")
+                ):
+                    local_path = Path(str(item.image.uri).replace("file:", ""))
+                    result.append(local_path)
+
+        return result
+
+    def _with_embedded_pictures(self) -> "DoclingDocument":
```

Comment:
[optional] this could be extended with an argument `fetch_remote=True`, which will download the remote urls